File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
let result = 0;
function calc(arr, index, current, target) {
    if(index === arr.length) {
        if(current === target) {
            result += 1;
        }
        return ;
    }
    //for(let i = index; i < arr.length-1; i++) {
    calc(arr, index+1, current+arr[index], target)
    calc(arr, index+1, current-arr[index], target)
    //}
}

function solution(numbers, target) {
    var answer = 0;
    // 1. 다음 원소를 더하거나 빼거나 경우
    // 2. 더하거나 뺀 결과 넘기기
    calc(numbers, 0, 0, target);
    return result;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
from collections import deque 

def bfs(numbers, target):
    queue = deque([])
    # 1. 부호는 두가지 경우 +, -
    queue.append(int(numbers[0]))
    queue.append(int(numbers[0])*(-1))
    # 2. 현재 값에 +, - 한 값을 더해서 넘긴다.
    
    
    for i in range(1, len(numbers)): 
        length = len(queue)
        for _ in range(length):
	        num = queue.popleft()
        	queue.append(num + int(numbers[i]))
        	queue.append(num + int(numbers[i])*(-1))
            
    target_count = [x for x in queue if x == target]
    return len(target_count)
        
def solution(numbers, target):
    answer = bfs(numbers, target)
    return answer